### PR TITLE
GSDocs: Allpix Squared Proposal

### PR DESCRIPTION
**Pull request by simonspa**
This is a project proposal for Google Season of Docs.

### CODE REVIEW

File: _gsdocs-projects/2020/project_AllpixSquared.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+---
+title: Allpix Squared
+project: AllpixSquared
+layout: default
+logo: AllpixSquared.png
```
Comment (xvallspl) [inline _gsdocs-projects/2020/project_AllpixSquared.md:5]: Image should be named following project-logo.png

File: _gsdocs-proposals/2020/proposal_AllpixSquaredRestructureDocs.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+* A restructured user manual which clearly separates content intended for users and developers
+* A reviewed manual with previously missing information on the project and the framework
+* A streamlined online presence which better integrates project website, documentation and code reference
+
+## Experience required
+Our current documentation workflow consists of a few tools and in order to rip them apart it would be good to have a solid understanding of:
+LaTeX, Doxygen, pandoc, hugo, git, GitLab and its Continuous Integration - as well as of all the new and cool tools that you will introduce to make our documentation sparkle.
+
+## Mentors
+  * [Paul Schütze](mailto:paul.schuetze@desy.de)
```
Comment (xvallspl) [inline _gsdocs-proposals/2020/proposal_AllpixSquaredRestructureDocs.md:56]: Please, update `gsdocs/2020/mentors.md` with the mentors

File: _gsdocs-projects/2020/project_AllpixSquared.md (hunk 1 above)
Comment (simonspa) [reply _gsdocs-projects/2020/project_AllpixSquared.md]: Done - had to also rename the file in old GSOC proposals though.

File: _gsdocs-proposals/2020/proposal_AllpixSquaredRestructureDocs.md (hunk 2 above)
Comment (simonspa) [reply _gsdocs-proposals/2020/proposal_AllpixSquaredRestructureDocs.md:56]: Done.

File: _gsdocs-proposals/2020/proposal_AllpixSquaredRestructureDocs.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+* **Infrastructure:** We would like to streamline the way we present the documentation and especially improve its online representation. Optimally we would end up with a streamlined, fully-integrated online presence which coherently presents news, manual and code reference. One possibility for this would e.g. be moving the documentation to [Docsy.dev](https://www.docsy.dev/). The challenging part about this is that we would like to keep as much as possible of our current automatic deployment and workflow.
+* **Bonus task:** For those of you who would like to dive a bit deeper into the scientific depth of this project - we currently do have a few examples for users to understand how to start using the framework and how to use certain features, but they are not very nicely presented and not very extensively documented. They could use a good polishing. Ultimately, these and further examples can be streamlined to follow a tutorial-like workflow.
```
Comment (agheata) [inline _gsdocs-proposals/2020/proposal_AllpixSquaredRestructureDocs.md:28]: Just a suggestion, part of your bonus task or something similar could be used as exercise to give to candidates, allowing you to select them (in addition to their portfolio).

File: _gsdocs-proposals/2020/proposal_AllpixSquaredRestructureDocs.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  * Repository of website (hugo): [APSQ Website @ CERn GitLab](https://gitlab.cern.ch/allpix-squared/allpix-squared-website)
+  * Project repository: [Allpix Squared @ CERN GitLab](https://gitlab.cern.ch/allpix-squared/allpix-squared)
+  * Documentation within the repository: [doc/usermanual](https://gitlab.cern.ch/allpix-squared/allpix-squared/-/tree/master/doc%2Fusermanual)
+  * Online documentation: [Online User Manual](https://project-allpix-squared.web.cern.ch/project-allpix-squared/usermanual/allpix-manual.html)
+  * PDF documentation: [PDF User Manual](https://project-allpix-squared.web.cern.ch/project-allpix-squared/usermanual/allpix-manual.pdf)
+  * Doxygen code reference: [Code Reference](https://project-allpix-squared.web.cern.ch/project-allpix-squared/reference)
+
+## Expected results
+
+* A restructured user manual which clearly separates content intended for users and developers
```
Comment (agheata) [inline _gsdocs-proposals/2020/proposal_AllpixSquaredRestructureDocs.md:47]: Why not a "User manual" and a "Developer guide", complementary to the Reference guide built with Doxygen? There are other projects having documentation split like this, e.g Geant4.

File: _gsdocs-projects/2020/project_AllpixSquared.md (hunk 1 above)
Comment (agheata) [reply _gsdocs-projects/2020/project_AllpixSquared.md]: In general we make a new copy if an old file was available - not to affect old pages. If you did it properly and grep'ed for the old file name and changed to the new name - it's fine I guess

File: _gsdocs-proposals/2020/proposal_AllpixSquaredRestructureDocs.md [hunk 5]
```diff
@@ -0,0 +1,57 @@
+---
+title: Generic Proposal Title
+layout: gsoc_proposal
```
Comment (agheata) [inline _gsdocs-proposals/2020/proposal_AllpixSquaredRestructureDocs.md:3]: The layout must read: gsdocs_proposal

File: _gsdocs-projects/2020/project_AllpixSquared.md (hunk 1 above)
Comment (simonspa) [reply _gsdocs-projects/2020/project_AllpixSquared.md]: I did - and even used `git mv` for renaming... :)

File: _gsdocs-proposals/2020/proposal_AllpixSquaredRestructureDocs.md (hunk 5 above)
Comment (simonspa) [reply _gsdocs-proposals/2020/proposal_AllpixSquaredRestructureDocs.md]: Ups, this I missed. It's also wrong in your example proposal, I fixed it there, too.

File: _gsdocs-proposals/2020/proposal_AllpixSquaredRestructureDocs.md (hunk 3 above)
Comment (simonspa) [reply _gsdocs-proposals/2020/proposal_AllpixSquaredRestructureDocs.md:28]: Is this something we need to state here or is it a bit like GSoC that we just send the evaluation task to the applicants who are interested?

File: _gsdocs-proposals/2020/proposal_AllpixSquaredRestructureDocs.md [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+
+* **Website:** The first point of contact should be the website. It is built using the static website generator `hugo` based on the very simple `beautiful-hugo` theme and contains, most importantly, the "news" section used for publishing release announcements and release notes. Furthermore, some installation information (mostly redundant to the user manual) and screenshots illustrating different use cases are provided.
+* **User Manual:** The main documentation of the framework consists of a 160-page user manual document which is generated from LaTeX code in the main software repository as well as Markdown files for individual parts of the framework - translated to LaTeX using `pandoc`. This happens automatically for new releases and it is published
+    * Online as HTML version with a link from the website (using `latex2html`)
+    * As PDF document, linked from the website (using `pdflatex`)
+    
+* **Doxygen Code Reference:** In addition to the user manual, also the source code is heavily documented and a HTML documentation of the code base is automatically generated using `doxygen` and deployed on the website.
```
[the remark this answers is not in the document]
Comment (simonspa) [reply _gsdocs-proposals/2020/proposal_AllpixSquaredRestructureDocs.md:22]: Check, thanks!

File: _gsdocs-proposals/2020/proposal_AllpixSquaredRestructureDocs.md [hunk 7]
```diff
@@ -0,0 +1,57 @@
+---
+title: Generic Proposal Title
```
[the remark this answers is not in the document]
Comment (simonspa) [reply _gsdocs-proposals/2020/proposal_AllpixSquaredRestructureDocs.md:2]: Ups... :grimacing:

File: _gsdocs-proposals/2020/proposal_AllpixSquaredRestructureDocs.md (hunk 4 above)
Comment (simonspa) [reply _gsdocs-proposals/2020/proposal_AllpixSquaredRestructureDocs.md:47]: We have discussed this but weren't entirely sure - the project isn't as big as Geant4 or others, and the development happens almost exclusively in close collaboration. Also, there is not so much information that a separate document is justified I guess.

Maybe something we can discuss with the candidate? I rephrased it accordingly.

File: _gsdocs-proposals/2020/proposal_AllpixSquaredRestructureDocs.md [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+    * Online as HTML version with a link from the website (using `latex2html`)
+    * As PDF document, linked from the website (using `pdflatex`)
+    
+* **Doxygen Code Reference:** In addition to the user manual, also the source code is heavily documented and a HTML documentation of the code base is automatically generated using `doxygen` and deployed on the website.
+
+With over 3.5 years of continuous development and additions, the documentation has become a bit cluttered and we would like to streamline it with your help:
+
+* **Content:** We would like to separate more clearly which parts of the documentation are intended for users which are not expected to alter a single line of code - and which parts aim at developers who would like to contribute to the project or would like to extend their own copy. Also, we would like to review the current user manual for potentially lacking information or areas where we could improve.
```
[the remark this answers is not in the document]
Comment (simonspa) [reply _gsdocs-proposals/2020/proposal_AllpixSquaredRestructureDocs.md:26]: Ah, this was indeed a bit misleading - I meant the copy of the software. I took your changes and clarified the second part a bit. Thanks!

File: _gsdocs-proposals/2020/proposal_AllpixSquaredRestructureDocs.md (hunk 3 above)
Comment (agheata) [reply _gsdocs-proposals/2020/proposal_AllpixSquaredRestructureDocs.md:28]: No, it was just a comment, no need to mention it in the proposal. Resolving conversation.

**Comment by agheata** [comment]
All good, thanks!

**Pull request merged**